### PR TITLE
Check Python version before install

### PR DESCRIPTION
**Pull request by pllim**
As suggested by @drdavella in https://github.com/astropy/astropy/issues/7306#issuecomment-374211168

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/env python
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-import os
 import sys
+
+if sys.version_info < (3, 5):
```
Comment (pllim) [inline setup.py]: Note: I got 3.5 from APE 10.

**Comment by bsipocz** [comment]
Yes, I agree with @mhvk, would be nice to not hard wire the version in the check, but put it down in a variable/config and use it in both places?

**Comment by pllim** [comment]
Ah, I didn't realize it is defined somewhere else. Thanks for the feedback. I'll apply the suggestion.

**Comment by drdavella** [comment]
It's going to be difficult to avoid duplicated code. We could put it in a config file, but note that in order to read the config file we need to import `astropy_helpers`, which is precisely one of the things we're trying to avoid if we have the wrong Python version.

**Comment by pllim** [comment]
Turns out, I can't import anything from `astropy/__init__.py` before `import ah_bootstrap`, where it is choking. I agree with @drdavella that having some duplication here is better than overly complicate things. If it helps, I will add a comment line to clarify where the duplication is for ease of future maintenance; is that okay?

**Comment by astrofrog** [comment]
+1 to just doing the simple thing even if it means duplication. Note that we might be able to reduce duplication with ``python_requires`` lower down in the setup.py file though? (maybe we can define the min version a single time at the top of setup.py).

**Comment by pllim** [comment]
@astrofrog , using `python_requires` in `setup()` does not prevent `import ah_bootstrap` from choking, no? @drdavella suggested moving the `sys` check to `ah_bootstrap.py` as well, but I am afraid putting it there would make it "hidden".

**Comment by astrofrog** [comment]
Sorry just to be clear, I'm suggesting avoiding hard-coding the same version twice in setup.py, but indeed we need the check at the start and also the python_requires setting in setup()

**Comment by astrofrog** [comment]
We need a similar check in ah_bootstrap:

https://github.com/astropy/astropy-helpers/issues/380

but that should stay separate because astropy-helpers and astropy don't *have* to support the same version ranges. But if you'd like to also do a PR to astropy-helpers to ah_bootstrap, that would be great :)

**Comment by drdavella** [comment]
Also, if this change is accepted here it should probably be added to the package template as well.

**Comment by pllim** [comment]
Is the template still using PY2-compatible helpers? Also, I am not sure how to open PR against the template anymore given this cookiecutter transition...

**Comment by drdavella** [comment]
@astrofrog yes as a placeholder with a comment, and maybe with some reasonable value filled in.

**Comment by mhvk** [comment]
OK to have duplicates, but do add a comment in both places that if one is updated, the other should as well.

**Comment by MSeifert04** [comment]
We can't use `distutils.version` here?

**Comment by drdavella** [comment]
@MSeifert04 I suppose we could, but it doesn't seem necessary. Using `sys.version` seems like a perfectly reasonable way to check the Python version. Unless I'm missing your point?

**Comment by MSeifert04** [comment]
yeah, definitively not a blocker, I just wondered.

**Comment by bsipocz** [comment]
Agreed with others above that it would be great to get this propagated back to the package template.

**Comment by pllim** [comment]
Since it is approved, merging then. I have opened two PRs for template and one more for helpers. :wink:

**Pull request merged**